crates/pixi_trampoline/src/main.rs

```diff
@@ -23,46 +22,78 @@ struct Metadata {
 fn read_metadata(current_exe: &Path) -> miette::Result<Metadata> {
     // the metadata file is next to the current executable parent folder,
     // under trampoline_configuration/current_exe_name.json
-    if let Some(exe_parent) = current_exe.parent(){
-        let metadata_path = exe_parent.join(TRAMPOLINE_CONFIGURATION).join(format!("{}{}", executable_from_path(current_exe), ".json"));
-        let metadata_file = File::open(&metadata_path).into_diagnostic().wrap_err(format!("Couldn't open {:?}", metadata_path))?;
+    if let Some(exe_parent) = current_exe.parent() {
+        let metadata_path = exe_parent.join(TRAMPOLINE_CONFIGURATION).join(format!(
+            "{}{}",
+            executable_from_path(current_exe),
+            ".json"
+        ));
+        let metadata_file = File::open(&metadata_path)
+            .into_diagnostic()
+            .wrap_err(format!("Couldn't open {:?}", metadata_path))?;
         let metadata: Metadata = serde_json::from_reader(metadata_file).into_diagnostic()?;
         return Ok(metadata);
     }
-    miette::bail!("Couldn't get the parent folder of the current executable: {:?}", current_exe);
+    miette::bail!(
+        "Couldn't get the parent folder of the current executable: {:?}",
+        current_exe
+    );
 }
 
-fn prepend_path(extra_path: &str) -> miette::Result<String> {
-    let path = env::var("PATH").into_diagnostic().wrap_err("Couldn't get 'PATH'")?;
-    let mut split_path = env::split_paths(&path).collect::<Vec<_>>();
-    split_path.insert(0, PathBuf::from(extra_path));
-    let new_path = env::join_paths(&split_path).into_diagnostic().wrap_err(format!("Couldn't join PATH's: {:?}", &split_path))?;
-    Ok(new_path.to_string_lossy().into_owned())
+/// Compute the difference between two PATH variables (the entries split by `;` or `:`)
+fn update_path(cached_path: String) -> String {
```

Comment:
I think it's better to receive here env_path and cached_path and compute just the difference.